src/lib/stores/filtered_items_store.ts

```diff
@@ -0,0 +1,30 @@
+import type { Filters } from 'src/types/filters.type';
+import type { Item } from 'src/types/item.type';
+import { writable } from 'svelte/store';
+
+let gItems: Item[];
+let gFilter: Filters;
+let itemDB: Item[];
+export let displayItems = writable<Item[]>(undefined);
+
+export const filterListener = (filter: Filters) => {
+	gFilter = filter;
+	filterItems();
+};
+
+export const itemsListener = (items: Item[]) => {
+	itemDB = items;
+	filterItems();
+};
```

Comment:
Ok. I see what you did here. It's basically a function that takes either the latest items or filters and stores globally. Then, it calls the filterItems. I assume the global storage was to save the latest state this had received in case the other was updated, right? 

You can achieve a VERY similar functionality with this: 

```ts
const displayItemsStore = derived([itemsStore, filterStore], ([items, filters], set) => {
    // Basically the code of `filterItems`, and then calling `set()` at the end instead of displayItems.set(). 
});
```